cpp/src/io/csv/csv_gpu.cu

```diff
@@ -402,6 +403,28 @@ __inline__ __device__ cudf::list_view decode_value(const char *data,
   return cudf::list_view{};
 }
 
+// The purpose of this is merely to allow compilation ONLY
+// TODO : make this work for csv
+template <>
+__inline__ __device__ numeric::decimal32 decode_value(const char *data,
+                                                      long start,
+                                                      long end,
+                                                      ParseOptions const &opts)
+{
+  return numeric::decimal32{};
```

Comment:
Can we not CUDF_FAIL here?

Reply:
I was pretty sure we couldn't `CUDF_FAIL` in device code and a double checked and sure enough got the following:
```
error: device code does not support exception handling
```

Reply:
Sorry didn't realize it was device code.